### PR TITLE
add Ruby 3.0 to CI matrix

### PR DESCRIPTION
**Pull request by robbkidd**
## Which problem is this PR solving?

- The absence of testing the latest version of Ruby.

## Short description of the changes

- Easiest to walk the commits, starting with [the first](https://github.com/honeycombio/beeline-ruby/pull/171/commits/e5fb055ef74fecc1e2a06356791007d19744e533). Each commit message describes a discrete fix for adding Ruby 3.0 to the matrix.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -124,7 +124,7 @@ workflows:
             - lint
           matrix:
             parameters:
-              ruby-version: ["2.2", "2.3", "2.4", "2.5", "2.6", "2.7"]
+              ruby-version: ["2.2", "2.3", "2.4", "2.5", "2.6", "2.7", "3.0"]
```
Comment (robbkidd) [inline .circleci/config.yml:127]: Note: Ruby 2.2 is included in the test matrix and is [the minimum version declared in the gemspec](https://github.com/honeycombio/beeline-ruby/blob/v2.7.0/honeycomb-beeline.gemspec#L18), but [the README says Ruby 2.3+ is required](https://github.com/honeycombio/beeline-ruby/blame/v2.7.0/README.md#L16). I'm inclined to let it ride for now.

**Review by vreynolds — APPROVED** [review]
onward

**Comment by robbkidd** [comment]
I'm gonna ship it!

**Comment by robbkidd** [comment]
Switched back to a patch bump and pending release because of the code changes to client-side integrations. This wasn't _only_ an update to the test suite, so those client-side changes should get released distinctly.

**Pull request merged**